### PR TITLE
add subdirs configuration parameter

### PR DESCRIPTION
**Pull request by kalefranz**

### CODE REVIEW

File: conda/base/context.py [hunk 1]
```diff
@@ -456,6 +461,7 @@ def list_parameters(self):
             'root_dir',
             'skip_safety_checks',
             'subdir',
+            'subdirs',
```
Comment (kalefranz) [inline conda/base/context.py]: These fields remain "hidden" and undocumented for now, until we're ready to expose them to users.

File: conda/core/index.py [hunk 2]
```diff
@@ -114,7 +115,7 @@ def fetch_index(channel_urls, use_cache=False, index=None):
 
     if index is None:
         index = {}
-    for _, repodata in repodatas:
+    for _, repodata in reversed(repodatas):
```
Comment (kalefranz) [inline conda/core/index.py]: This is a major change in logic, but only affects the contents of `index` if there's a hash collision.  When there *is* a hash collision, this is actually the correct logic.

File: conda/models/channel.py [hunk 3]
```diff
@@ -391,17 +395,19 @@ def url(self, with_credentials=False):
         return None
 
 
-def prioritize_channels(channels, with_credentials=True, platform=None):
+def prioritize_channels(channels, with_credentials=True, subdirs=None):
```
Comment (kalefranz) [inline conda/models/channel.py]: Here we're changing the API for `prioritize_channels()` as well as the `urls()` method on the `Channel` object.  I don't see `prioritize_channels` in exports, so I'm hopeful we'll be ok here with conda-build.  I guess we'll see when the tests run.

File: conda/core/index.py [hunk 4]
```diff
@@ -88,7 +88,8 @@ def get_index(channel_urls=(), prepend=True, platform=None,
     if context.offline and unknown is None:
         unknown = True
 
-    channel_priority_map = prioritize_channels(channel_urls, platform=platform)
+    subdirs = (platform, 'noarch') if platform is not None else context.subdirs
+    channel_priority_map = prioritize_channels(channel_urls, subdirs=subdirs)
```
Comment (kalefranz) [inline conda/core/index.py]: This is the `get_index()` function.  The API remains the same right now.  You can see I'm translating the `platform` argument into `subdirs`.  At some point, we could switch out a `subdirs` for `platform` if we wanted.

File: conda/models/channel.py (hunk 3 above)
Comment (msarahan) [inline conda/models/channel.py]: I think it'll be fine.  We'll see though.

**Pull request merged**